agent/json_utils.py

```
@@ -0,0 +1,49 @@
+    clean = re.sub(r"```[a-zA-Z]*\s*\n?", "", clean)
+    clean = re.sub(r"\n?\s*```", "", clean)
+    clean = clean.strip()
+
+    start = clean.find('{')
+    if start == -1:
+        raise ValueError(f"No JSON object found: {clean[:120]}")
+
+    depth, in_str, esc = 0, False, False
+    for i, ch in enumerate(clean[start:], start):
+        if esc:
+            esc = False; continue
```

Ruff (0.15.20)

[error] Multiple statements on one line (semicolon)

(E702)